src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,403 @@
+                                            DetailNode moduleJavadoc)
+            throws MacroExecutionException {
+        final Field field = SiteUtil.getField(instance.getClass(), propertyName);
+
+        sink.rawText(INDENT_LEVEL_12);
+        sink.tableRow();
+
+        writePropertyNameCell(sink, propertyName);
+        writePropertyDescriptionCell(sink, propertyName, propertyJavadoc);
+        writePropertyTypeCell(sink, propertyName, field, instance);
+        writePropertyDefaultValueCell(sink, propertyName, field, instance);
+        writePropertySinceVersionCell(
+                sink, propertyName, moduleJavadoc, propertyJavadoc);
+
+        sink.rawText(INDENT_LEVEL_12);
+        sink.tableRow_();
+    }
+
+    /**
+     * Writes a table cell with the given property name.
+     *
+     * @param sink sink to write to.
+     * @param propertyName the name of the property.
+     */
+    private static void writePropertyNameCell(Sink sink, String propertyName) {
+        sink.rawText(INDENT_LEVEL_14);
+        sink.tableCell();
+        sink.text(propertyName);
+        sink.tableCell_();
+    }
+
+    /**
+     * Writes a table cell with the property description.
+     *
+     * @param sink sink to write to.
+     * @param propertyName the name of the property.
+     * @param propertyJavadoc the Javadoc of the property containing the description.
+     * @throws MacroExecutionException if an error occurs during retrieval of the description.
+     */
+    private static void writePropertyDescriptionCell(Sink sink, String propertyName,
+                                                     DetailNode propertyJavadoc)
+            throws MacroExecutionException {
+        sink.rawText(INDENT_LEVEL_14);
+        sink.tableCell();
+        final String description = SiteUtil
+                .getPropertyDescription(propertyName, propertyJavadoc, currentModuleName);
+        sink.rawText(description);
+        sink.tableCell_();
+    }
+
+    /**
+     * Writes a table cell with the property type.
+     *
+     * @param sink sink to write to.
+     * @param propertyName the name of the property.
+     * @param field the field of the property.
+     * @param instance the instance of the module.
+     * @throws MacroExecutionException if link to the property_types.html file cannot be
+     *                                 constructed.
+     */
+    private static void writePropertyTypeCell(Sink sink, String propertyName,
+                                              Field field, Object instance)
+            throws MacroExecutionException {
+        sink.rawText(INDENT_LEVEL_14);
+        sink.tableCell();
+
+        if (SiteUtil.TOKENS.equals(propertyName)) {
+            final AbstractCheck check = (AbstractCheck) instance;
+            final List<Integer> tokens = SiteUtil
+                    .subtractTokens(check.getAcceptableTokens(), check.getRequiredTokens());
```

Comment:
The list is displaying tokens that users can customize/personalize for their own needs. We subtract "required" tokens because they can't be customized and are mandatory to always be included whether the user lists them or not.

Reply:
Ah, I see. Let's rename this to `configurabletokens` to make this obvious.

Reply:
Done.